src/ray/gcs/redis_async_context.cc

```diff
@@ -0,0 +1,51 @@
+#include "ray/gcs/redis_async_context.h"
+
+extern "C" {
+#include "ray/thirdparty/hiredis/async.h"
+#include "ray/thirdparty/hiredis/hiredis.h"
+}
+
+namespace ray {
+
+namespace gcs {
+
+Status RedisAsyncContext::RedisAsyncCommand(redisCallbackFn *fn, void *privdata,
+                                            const char *format, ...) {
+  va_list ap;
+  va_start(ap, format);
+
+  int ret_code = 0;
+  {
+    std::lock_guard<std::mutex> lock(mutex_);
```

Comment:
Maybe add some comments about why the locks are necessary?

Reply:
Sounds good.